File: utils/lib-core/src/main/java/com/yyxnb/what/core/log/LogConfig.java
package com.yyxnb.what.core.log;

import android.text.TextUtils;

import com.yyxnb.what.app.AppUtils;


public class LogConfig {

    private boolean showThreadInfo = true;
    private boolean debug = AppUtils.isDebug();
    private String tag = "----arch----";


    public LogConfig setTag(String tag) {
        if (!TextUtils.isEmpty(tag)) {
            this.tag = tag;
        }
        return this;
    }

    public LogConfig setShowThreadInfo(boolean showThreadInfo) {
        this.showThreadInfo = showThreadInfo;
        return this;
    }


    public LogConfig setDebug(boolean debug) {
        this.debug = debug;
        return this;
    }

    public String getTag() {
        return tag;
    }

    public boolean isDebug() {
        return debug;
    }

    public boolean isShowThreadInfo() {
        return showThreadInfo;
    }
}

File: arch/src/main/java/com/yyxnb/arch/utils/log/LogConfig.kt
package com.yyxnb.arch.utils.log


import android.text.TextUtils


class LogConfig {

    private var showThreadInfo = true
    private var debug = isDebug()
    private var tag = "arch"


    fun setTag(tag: String): LogConfig {
        if (!TextUtils.isEmpty(tag)) {
            this.tag = tag
        }
        return this
    }

    fun setShowThreadInfo(showThreadInfo: Boolean): LogConfig {
        this.showThreadInfo = showThreadInfo
        return this
    }


    fun setDebug(debug: Boolean): LogConfig {
        this.debug = debug
        return this
    }

    fun getTag(): String {
        return tag
    }

    fun isDebug(): Boolean {
        return debug
    }

    fun isShowThreadInfo(): Boolean {
        return showThreadInfo
    }
}
